modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/core/AttributionDocumentGeneratorImpl.java

```diff
@@ -72,15 +77,41 @@ public AttributionDocumentGeneratorImpl(String documentName, File workingDir, St
      * @return (non - null) the file handle of the generated attribution document.
      */
     public File generate(List<ArtifactAndLicense> artifacts) {
-        Templates templates = TemplateLoaderUtil.load(templateKey);
-
-        File title = writeTitle(templates);
-        File copyright = writeCopyright(templates);
-        File backPage = writeBackPage(templates.getBackPage());
-        File artifactPages = writeArtifacts(templates, artifacts, templates.getContent());
-        File intermediateDoc = mergePages(title, copyright, artifactPages, backPage);
-
-        return postProcess(templates, intermediateDoc, documentName);
+        Templates templates = null;
+        try {
+            templates = TemplateLoaderUtil.load(templateKey);
+
+            File title = writeTitle(templates);
+            File copyright = writeCopyright(templates);
+            File backPage = writeBackPage(templates.getBackPage());
+            File artifactPages = writeArtifacts(templates, artifacts, templates.getContent());
+            File intermediateDoc = mergePages(title, copyright, artifactPages, backPage);
+
+            return postProcess(templates, intermediateDoc, documentName);
+        } finally {
+            if (templates != null) {
+                try {
+                    templates.getTitle().close();
+                } catch (IOException e) {
+                    LOG.warn("Failed to close the PDDocument object.", e);
```

Comment:
One small improvement here, the warning should not contain the exception, but should refer to the document that could not be closed. In addition, there should be a debug log with the exception. A normal user should not be confronted with exceptions, but in debug log, enough information should be given.